src/config.rs

```diff
@@ -158,118 +160,221 @@ impl From<NoiseSuppressionLevel> for ffi::NoiseSuppression_SuppressionLevel {
+            enabled: true,
+            mode: other.mode.into(),
+            target_level_dbfs: other.target_level_dbfs as i32,
+            compression_gain_db: other.compression_gain_db as i32,
+            enable_limiter: other.enable_limiter,
+            ..Default::default()
         }
     }
 }
 
-/// Config that can be used mid-processing.
+/// The parameters and behavior of the audio processing module are controlled
+/// by changing the default values in this `Config` struct.
+/// The config is applied by passing the struct to the [`set_config`] method.
 #[derive(Debug, Default, Clone, PartialEq)]
 #[cfg_attr(feature = "derive_serde", derive(Serialize, Deserialize))]
 pub struct Config {
-    /// Enable and configure AEC (acoustic echo cancellation).
-    pub echo_cancellation: Option<EchoCancellation>,
+    /// Sets the properties of the audio processing pipeline.
+    pub pipeline: Pipeline,
+
+    /// Enables and configures the pre-amplifier. It amplifies the capture signal before any other
+    /// processing is done.
+    pub pre_amplifier: Option<PreAmplifier>,
+
+    /// Enables and configures high pass filter.
+    pub high_pass_filter: Option<HighPassFilter>,
 
-    /// Enable and configure AGC (automatic gain control).
-    pub gain_control: Option<GainControl>,
+    /// Enables and configures acoustic echo cancellation.
+    pub echo_canceller: Option<EchoCanceller>,
 
-    /// Enable and configure noise suppression.
+    /// Enables and configures background noise suppression.
     pub noise_suppression: Option<NoiseSuppression>,
 
-    /// Enable and configure voice detection.
-    pub voice_detection: Option<VoiceDetection>,
+    /// Enables transient noise suppression.
+    pub enable_transient_suppression: bool,
 
-    /// Use to enable experimental transient noise suppression.
-    pub enable_transient_suppressor: bool,
+    /// Enables reporting of [`voice_detected`] in [`Stats`].
+    pub enable_voice_detection: bool,
 
-    /// Use to enable a filtering component which removes DC offset and
-    /// low-frequency noise.
-    pub enable_high_pass_filter: bool,
+    /// Enables and configures automatic gain control.
+    /// TODO: Experiment with and migrate to GainController2.
+    pub gain_controller: Option<GainController>,
+
+    /// Enables reporting of [`residual_echo_likelihood`] and
+    /// [`residual_echo_likelihood_recent_max`] in [`Stats`].
+    pub enable_residual_echo_detector: bool,
+
+    /// Enables reporting of [`output_rms_dbfs`] in [`Stats`].
+    pub enable_level_estimation: bool,
 }
 
-impl From<Config> for ffi::Config {
-    fn from(other: Config) -> ffi::Config {
-        let echo_cancellation = if let Some(enabled_value) = other.echo_cancellation {
-            enabled_value.into()
+impl From<Config> for ffi::AudioProcessing_Config {
+    fn from(other: Config) -> Self {
+        let pre_amplifier = if let Some(config) = other.pre_amplifier {
+            config.into()
         } else {
```

Comment:
Yeah, I sort of like that each field has the explicit `enabled: false`, esp because it's a FFI struct that doesn't follow Rust conventions. I think I will leave it as is?